### PR TITLE
build: Add infrastructure to setup an account for the first time

### PR DESCRIPTION
**Pull request by nikiburggraf**
Contains the following

* Account setup infra
* Commands for adding/removing/listing/etc users that can assume the deploy and ops roles
* Some refactoring (everything moved under deployment/cloud/aws, setups split between `account` and `infrastructure`, ops roles split out of the server module and into their own module)

This ended up being a monster, I'm really sorry about that!

### CODE REVIEW

File: deployment/cloud/aws/account/main.tf [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/AmazonEC2FullAccess",
+    "arn:aws:iam::aws:policy/AmazonVPCFullAccess",
+    "arn:aws:iam::aws:policy/AutoScalingFullAccess",
+    "arn:aws:iam::aws:policy/IAMFullAccess",
```
Comment (lsetiawan) [inline deployment/cloud/aws/account/main.tf:73]: Does this deploy user need an IAM Full Access? They're not messing around with IAM at the deployment stage right?

File: deployment/cloud/aws/account/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,5 @@
+account_resource_prefix = "supportsphere"
+account_additional_tags = {
+  "Project" = "Support Sphere"
+}
+account_id = "871683513797"
```
Comment (lsetiawan) [inline deployment/cloud/aws/account/terraform.tfvars:5]:
```suggestion
account_id = "871683513797"
ops_group_name = "ssec-eng"
```

File: deployment/cloud/aws/infrastructure/main.tf [hunk 3]
```diff
@@ -45,22 +47,50 @@ provider "aws" {
...
-module "server" {
-  source = "./modules/server"
+module "keys" {
+  source = "./modules/keys"
   
   resource_prefix = local.resource_prefix
-  instance_type = var.instance_type
   stage = var.stage
-  volume_size = var.volume_size
+  ops_group_name = "ssec-eng"
```
Comment (lsetiawan) [inline deployment/cloud/aws/infrastructure/main.tf:66]:
```suggestion
  ops_group_name = var.ops_group_name
```

File: deployment/cloud/aws/infrastructure/main.tf [hunk 4]
```diff
@@ -45,22 +47,50 @@ provider "aws" {
...
+  kms_key_arn_west = module.keys.kms_key_arn_west
+  kms_key_arn_east = module.keys.kms_key_arn_east
+}
+
+module "ops-roles" {
+  source = "./modules/ops-roles"
+  
+  resource_prefix = local.resource_prefix
+  stage = var.stage
+  ops_group_name = "ssec-eng"
```
Comment (lsetiawan) [inline deployment/cloud/aws/infrastructure/main.tf:90]:
```suggestion
  ops_group_name = var.ops_group_name
```

**Review by lsetiawan — COMMENTED** [review]
I added some suggested changes above. Not sure if the ops group name was intentionally hardcoded in some places. I think otherwise, everything seem reasonable. Definitely looks like a lot of work has been put in! Nice job in figuring all of this out. It's quite valuable and the structure looks good. 👍🏽 I'll hold on approving for a bit until my questions are answered. Thanks.

File: deployment/cloud/aws/account/main.tf (hunk 1 above)
Comment (nikiburggraf) [reply deployment/cloud/aws/account/main.tf:73]: They need to create ops roles & associate their permissions with the `ssec-eng` group, instance policies, policies encrypt/decrypt access for the SOPS keys, etc. This could definitely be scoped down (which I do with some `deny` roles on all the IAM User-related-stuff further down here)

tl;dr -- they need some IAM access at the deploy stage

**Comment by nikiburggraf** [comment]
> Not sure if the ops group name was intentionally hardcoded in some places.

Not intentionally! Added the variable later on and I guess I didn't fully clean up all the hardcoded references!

**Review by lsetiawan — APPROVED** [review]
This looks good. Thank you :shipit:
